index.d.ts

```diff
@@ -1,7 +1,12 @@
+type MethodTypes = 'widget-opened' | 'widget-closed' | 'error' | 'workflow-finished'
+
 interface WidgetApi {
-    open: () => void
+    open: () => void;
+    close: () => void;
     setContext: (p: string) => void;
     transactionlink_ready: () => void;
+    on: (event: MethodTypes , cb: () => void) => void;
```

Comment:
temporarily using something like the code below will allow you to fill in the type of callback parameter that is now missing
```
for (const event of [
        'error',
        'widget-opened',
        'widget-closed',
        'app-opened',
        'app-closed',
        'page-changed',
        'workflow-finished'
    ]) {
        window.transactionlink.on(event, (e) => {
            console.warn(`"${event}": `, JSON.stringify(e))
        })
    }
```

Reply:
okay thanks, I will check the type of parameter in callback